### PR TITLE
renders list of articles, can sort by topic

### PR DESCRIPTION
**Pull request by Superjim**

### CODE REVIEW

**Review by Superjim — COMMENTED** [review]
im happy with this

File: src/components/Topic.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    };
+
+    fetchArticles();
+  }, [topic_slug]);
+  return (
+    <div>
+      {topic_slug ? <h2>Topic {topic_slug}</h2> : <h2>All Topics</h2>}
+      <ul>
+        {articles.map((article) => (
+          <li key={article.article_id}>{article.title}</li>
```
Comment (Hard-Luck) [inline src/components/Topic.jsx:25]: I would have each one as an <Article /> card rather than a generic li

File: src/components/Topic.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { useParams } from "react-router-dom";
+import axios from "axios";
+
+function Topic() {
+  let { topic_slug } = useParams();
+  if (topic_slug === undefined) topic_slug = "";
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    const fetchArticles = async () => {
```
Comment (Hard-Luck) [inline src/components/Topic.jsx:11]: This logic should be abstracted into a utils file

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Extract both pieces of logic from the useEffect into utils and create an Article component.

Other than that good work!

**Pull request merged**